### PR TITLE
Intial Upload

### PR DESCRIPTION
**Pull request by grameenamerica**

### CODE REVIEW

File: Canary/Canary/App.config [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
+  </startup>
+
+  <appSettings>
+
+    <add key="MambuIp" value="52.201.89.67"/>
+    <add key="SecurityKey" value=="{SecurityKey}"/>
+    <add key="MambuUrl" value="https://grameenamerica.sandbox.mambu.com"/>
+    <add key="ApiUerId" value="{ApiUerId}"/>
```
Comment (davestern) [inline Canary/Canary/App.config]: Is this a typo? `ApiUerId` or `ApiUserId`?

**Comment by davestern** [comment]
@grameenamerica this is a great start. A few things about this code:

1. There are several keys and tokens in the code that look like they should be removed. Please check these. This may also apply to phone numbers in Twilio.
1. There are several "Debug" files here. If they serve no purpose other than debugging, they should be removed.
1. Can you please look through the rest of the code to make sure secrets, tokens or keys that shouldn't be there are removed or replaced by template variables?

File: Canary/Canary/App.config (hunk 1 above)
Comment (grameenamerica) [reply Canary/Canary/App.config]: This is a typo. But I have used the same key inside the code.

**Comment by grameenamerica** [comment]
> @grameenamerica this is a great start. A few things about this code:
> 
> 1. There are several keys and tokens in the code that look like they should be removed. Please check these. This may also apply to phone numbers in Twilio.
> 2. There are several "Debug" files here. If they serve no purpose other than debugging, they should be removed.
> 3. Can you please look through the rest of the code to make sure secrets, tokens or keys that shouldn't be there are removed or replaced by template variables?

Hi Dave,
Can I add these changes and commit again?

**Comment by davestern** [comment]
Yes, you absolutely should. The PR is a work in progress. It's mean to be
collaborative. So you can commit, push the commits and ask me to review
(you can assign me as a reviewer in Github and I'll get a notification).
I'll make comments, you make changes and commit & push again. This is a
Code Review and it's standard procedure in most shops. Once we're both
satisfied that it's production ready and tested, we merge into the master
branch and deploy.

File: Canary/Canary/App.config (hunk 1 above)
Comment (davestern) [reply Canary/Canary/App.config]: OK, feel free to update in the code if you want, but not a blocker

Comment (grameenamerica) [reply Canary/Canary/App.config]: I have corrected the typo. Please check and let me know.

File: Canary/Canary/Program.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+                return;
+            }
+        }
+
+        static void SendText2()
+
+        {
+            var accountSid = "ACb637647aa631592e3795f8fcba45f386";
+            var authToken = "40d2b23aed782684116da937a4faf732";
```
[the remark this answers is not in the document]
Comment (grameenamerica) [reply Canary/Canary/Program.cs]: I have removed all account information.

File: Canary/Canary/Program.cs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+                textBody = "SANDBOX - Something might went wrong on webhook";
+            }
+
+
+            try
+            {
+
+                //string memberPhNumber = "+16463097471";
+                string memberPhNumber = "+18478675512";
```
[the remark this answers is not in the document]
Comment (grameenamerica) [reply Canary/Canary/Program.cs]: Phone numbers are removed from the code. They are added into the config file.

File: Canary/Canary/bin/Debug/Canary.vshost.exe.config [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
+  </startup>
+  <appSettings>
+    <add key="MambuIp" value="52.201.89.67" />
+    <add key="SecurityKey" value="da39a3ee5e6b4b0d3255bfef95601890afd80709" />
```
[the remark this answers is not in the document]
Comment (grameenamerica) [reply Canary/Canary/bin/Debug/Canary.vshost.exe.config]: Security key is removed but I have left IP address as it is as I believe this is not private information.

File: Canary/Canary/Program.cs [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+            return minutes;
+        }
+
+
+
+        static void SendText1()
+
+        {
+            var accountSid = "ACb637647aa631592e3795f8fcba45f386";
+            var authToken = "40d2b23aed782684116da937a4faf732";
```
[the remark this answers is not in the document]
Comment (grameenamerica) [reply Canary/Canary/Program.cs]: This is also removed to config file.

File: Canary/Canary/bin/Debug/Canary.vshost.exe.config [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
+  </startup>
+
+  <appSettings>
+
+    <add key="MambuIp" value="52.201.89.67"/>
+    <add key="SecurityKey" value=="{SecurityKey}"/>
+    <add key="MambuUrl" value="https://grameenamerica.sandbox.mambu.com"/>
+    <add key="ApiUerId" value="{ApiUerId}"/>
```
Comment (davestern) [inline Canary/Canary/bin/Debug/Canary.vshost.exe.config]: Looks like you missed one @grameenamerica

File: Canary/Canary/Program.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (davestern) [reply Canary/Canary/Program.cs]: Did you push your change? I still see it here

File: Canary/Canary/Program.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (davestern) [reply Canary/Canary/Program.cs]: Did you push your change? I still see it here

**Comment by davestern** [comment]
@grameenamerica I'm not sure you got all the tokens and secrets out of your code. Did you forget to push changes or do you need to do another pass?

File: Canary/Canary/Program.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (grameenamerica) [reply Canary/Canary/Program.cs]: Yes I did. I don't see Sid or Token from my side. I can see this.

Capture

File: Canary/Canary/Program.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (grameenamerica) [reply Canary/Canary/Program.cs]: I pushed the changes. I do not see any security info, phone numbers when I open the program.cs file. Also I can see the new lines in the app.config file.

File: Canary/Canary/bin/Debug/Canary.vshost.exe.config (hunk 6 above)
Comment (grameenamerica) [reply Canary/Canary/bin/Debug/Canary.vshost.exe.config]: MambuIP and MambuUrl is removed

File: Canary/Canary/obj/Debug/Canary.csproj.Canary.exe.config [hunk 7]
```diff
@@ -0,0 +1,21 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
+  </startup>
+  <appSettings>
+    <add key="MambuIp" value="52.201.89.67" />
+    <add key="SecurityKey" value="da39a3ee5e6b4b0d3255bfef95601890afd80709" />
```
[the remark this answers is not in the document]
Comment (grameenamerica) [reply Canary/Canary/obj/Debug/Canary.csproj.Canary.exe.config]: this is removed

File: Canary/Canary/Program.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (davestern) [reply Canary/Canary/Program.cs]: I see that now as well.

**Pull request merged**